test/fuzz/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+# Copyright (C) 2022-2024 Intel Corporation
+# Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+message(STATUS "Fuzzing tests enabled.")
+
+function(add_fuzz_test name label)
+    add_test(
+        NAME "fuzz-${name}"
+        COMMAND fuzztest ${ARGN} -verbosity=1
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
+    set_tests_properties("fuzz-${name}" PROPERTIES LABELS ${label})
+endfunction()
+
+set(TEST_TARGET_NAME fuzztest)
+add_umf_executable(
+    NAME ${TEST_TARGET_NAME}
+    SRCS umfFuzz.cpp
+    LIBS umf)
+target_link_libraries(${TEST_TARGET_NAME} PRIVATE umf -fsanitize=fuzzer)
+target_compile_options(${TEST_TARGET_NAME} PRIVATE -g -fsanitize=fuzzer)
```

Comment:
Is fuzzer always available on clang? Perhaps, we should check for this flag availability using `check_c/cxx_compiler_flag`, similar to how it's done here: https://github.com/oneapi-src/unified-memory-framework/blob/ddfb338cc061103ff9aa144c93aa759f70f7b184/cmake/helpers.cmake#L250.

Reply:
Such a check didn't work for me locally, even though fuzzing works. Perhaps worth investigating in another PR.

Reply:
I think it is worth investigating in another PR